job-service-db/src/main/resources/db/migration/functions/public/R__createJobForDependencies.sql

```diff
@@ -138,6 +138,20 @@ BEGIN
         SELECT job_id FROM prereqs_created_but_not_complete
         UNION
         SELECT job_id FROM prereqs_not_created_yet
+        -- The SELECT query below is only used for its side effects, which is to call the 'internal_raise_exception_for_failed_prereqs'
+        -- function (its not enough to just add a new CTE to this pipeline that calls the
+        -- 'internal_raise_exception_for_failed_prereqs' function, as that new CTE must also be used in a later query, otherwise the
+        -- 'internal_raise_exception_for_failed_prereqs' won't get called).
+        -- If any prereqs have a 'Failed' status, the SELECT query will raise an exception and terminate this function ('create_job').
+        -- If no prereqs have a 'Failed' status, the SELECT query won't return anything.
+        UNION
+        SELECT job_id
+        FROM (SELECT uj.job_id, internal_raise_exception_for_failed_prereqs(array_agg(uj.job_id)) AS failed_prereqs
+              FROM updated_jobs uj
+              WHERE uj.partition_id = in_partition_id
+                AND uj.job_id IN (SELECT job_id FROM prereqs)
+                AND uj.status = 'Failed'
+              GROUP BY uj.job_id) AS failed_jobs
```

Comment:
Let's discuss this.  I don't understand the need to introduce arrays for this, and also there are a few things I don't understand about it.